varinfo/cmr_search.py

```diff
@@ -0,0 +1,66 @@
+''' This script leverages the python-cmr library to conduct a CMR search
+'''
+# Import CMR query and environment types
+from cmr import GranuleQuery, CMR_OPS, CMR_UAT, CMR_SIT
+
+def query_cmr(concept_id: str = None, sn: str = None, 
+             ver: str = None, provider: str = None,
+             page_size: int = 10, env: CMR_OPS = CMR_OPS) -> list:
+    ''' 
+    Search CMR given:
+        * concept_id: a collection or granule's IDs
+        * sn/short_name: 'SNDRSNIML2CCPRET' or 'M2T1NXSLV'
+        * ver/version: a collection version (e.g. '2' or 5.12.4')
+        * provider: data center ID (e.g. GES_DISC, PODAAC, POCLOUD, EEDTEST)
+        * page_size: number of links to return, the default is 10
+        * env/environment: CMR environments (OPS, UAT, and SIT)
+    '''
+    granule_query = GranuleQuery(mode=env)
+    
+    if concept_id: # Check for concept_id
+        # Set `sort_key` = `-start_date` to get most recent granules first
+        query_parameters = {'concept_id': concept_id,
+                            'sort_key': '-start_date'}
+    
+    elif sn and ver and provider: 
+        # Check for short_name, version, and provider
+        query_parameters = {'short_name': sn, 
+                            'version': ver, 
+                            'provider': provider,
+                            'sort_key': '-start_date'}
+    else:
+        # If no short_name, version and provider raise TypeErorr
+        raise TypeError(
```

Comment:
I think it's a `TypeError` because of this: https://docs.python.org/3/library/exceptions.html#TypeError

Reply:
Hmmm, it sounds like `CustomException` time? Can I just use the `CustomError` exception and than add and error message or should I make a new class specifically for this?

Reply:
We chatted about this in Slack, but this would be a new custom exception (another child of the `CustomError` class in `exception.py`, similar to what you are doing for the CMR query exception handling.